SU2_PY/SU2/eval/design.py

```diff
@@ -364,7 +364,7 @@ def obj_df(dvs,config,state=None):
             # For multiple objectives are evaluated one-by-one rather than combined
             # MARKER_MONITORING should be updated to only include the marker for i_obj
             # For single objectives, multiple markers can be used 
-            config['MARKER_MONITORING'] = marker_monitored[i_obj]
+            if (n_obj>1): config['MARKER_MONITORING'] = marker_monitored[i_obj]
```

Comment:
I think that the array of monitored markers is extended to accommodate this situation in config.py (~line 514), and there is a single objective-with-multiple-surfaces regression test; do you have a case you can send me where this wasn't working? Or if you see an issue with that test that I previously missed I wouldn't mind looking into it.

Reply:
@fpalacios yes, I commented a bit too quickly; I do see the behavior that you mention.  I think the line previously mentioned helps with getting the value of the objective but not the gradient.
While I was testing this on my end I also noticed that there is an issue if you list the same objective multiple times (for example, if you wanted to weight the drag contribution from one surface more than another), where since it has the same key the previous dict entry is overwritten. Do you mind if I add an error catching that to your PR?

I am taking a look back through it but so far I think your solution will be good - and better to fix the performance in master now and worry about elegant improvements in the develop branch.